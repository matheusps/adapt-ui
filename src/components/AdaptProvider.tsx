import React, { FC } from 'react'
import { ThemeProvider, DefaultTheme } from 'styled-components'

import { theme } from '../config'
import { GlobalStyle } from './GlobalStyle'

interface Props {
  theme?: DefaultTheme
}

/**
 * Wraps around Theme-provider from emotion üíÖüèª
 *
 */
const AdaptProvider: FC<Props> = ({ theme, children }) => {
  return (
    <ThemeProvider theme={theme!}>
      <>
        <GlobalStyle />
        {children}
      </>
    </ThemeProvider>
  )
}

AdaptProvider.defaultProps = {
  theme: theme,
}

export default AdaptProvider
