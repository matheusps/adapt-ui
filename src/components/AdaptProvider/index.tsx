import React, { FC } from 'react'
import { ThemeProvider, DefaultTheme } from 'styled-components'

import { theme } from '../../config'
import GlobalPreflight from './GlobalPreflight'
import TypoPreflight from '../Typography/Preflight'

interface Props {
  theme?: DefaultTheme
}

const AdaptProvider: FC<Props> = ({ theme, children }) => {
  return (
    <ThemeProvider theme={theme!}>
      <>
        <GlobalPreflight />
        <TypoPreflight />
        {children}
      </>
    </ThemeProvider>
  )
}

AdaptProvider.defaultProps = {
  theme: theme,
}

export default AdaptProvider
