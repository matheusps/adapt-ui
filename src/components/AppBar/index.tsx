import React, { FC } from 'react'
import Surface from '../Surface'
import Title from './Title'

interface Props extends HasSkin {
  fixed?: boolean
  title?: string
  theme: string
}

/**
 * TODO: Component meaning
 */
const AppBar: FC<Props> = ({ fixed, skin, title, children }) => {
  return (
    <Surface>
      {title && <Title text={title} />}
      {children}
    </Surface>
  )
}

AppBar.defaultProps = {
  skin: 'primary',
}

export default AppBar
