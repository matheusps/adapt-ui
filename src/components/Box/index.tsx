import styled from 'styled-components'
import styleComposer from '../../styleComposer'

const defaultSpacing = {
  scale: {
    sm: '5rem',
    md: '2rem',
    lg: '3rem',
  },
  propName: {
    margin: 'm',
    marginTop: 'mt',
    marginRight: 'mr',
    marginBottom: 'mb',
    marginLeft: 'ml',
    padding: 'p',
    paddingTop: 'pt',
    paddingRight: 'pr',
    paddingBottom: 'pb',
    paddingLeft: 'pl',
  },
}

const defaultSizes = {
  scale: {
    sm: '50%',
    md: '2rem',
    lg: '3rem',
  },
  propName: {
    width: 'width',
    height: 'height',
  },
}

const spacing = styleComposer(defaultSpacing) as Spacing
const sizes = styleComposer(defaultSizes) as Sizes

const Box = styled.div<BoxProps>`
  ${spacing}
  ${sizes}
  box-sizing: border-box;
`

type Scale = 'sm' | 'md' | 'lg'
type Spacing = {
  m?: Scale
  mt?: Scale
  mr?: Scale
  mb?: Scale
  ml?: Scale
  p?: Scale
  pt?: Scale
  pr?: Scale
  pb?: Scale
  pl?: Scale
}

type Sizes = {
  height?: Scale
  width?: Scale
}

export type BoxProps = Spacing & Sizes

export default Box
