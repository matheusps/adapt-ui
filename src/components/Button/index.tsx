import { get } from 'lodash'
import styled from 'styled-components'

import {
  getMeasure,
  lightenOrDarken,
  isBrightColor,
} from '../../global/helpers'

interface Props extends withGoal {
  size?: Measure
  kind?: 'link' | 'ghost' | 'default'
}

const getFontSize = (measure: Measure) =>
  getMeasure(measure, ['0.8rem', '1rem', '1.25rem', '1.50rem'])

const getPadding = (measure: Measure) =>
  getMeasure(measure, [
    '0.375rem 1.4rem 0.375rem 1.4rem',
    '0.45rem 1.8rem 0.45rem 1.8rem',
    '0.65rem 2.2rem 0.65rem 2.2rem',
    '0.80rem 2.6rem 0.80rem 2.6rem',
  ])

const Button = styled.button<Props>`
  background-color: ${({ kind, theme, goal }) =>
    kind === 'ghost' || kind === 'link'
      ? 'transparent'
      : get(theme.colors.goal, goal)};
  color: ${({ kind, theme, goal }) =>
    kind === 'ghost' || kind === 'link'
      ? get(theme.colors.goal, goal)
      : isBrightColor(get(theme.colors.goal, goal))
      ? '#000'
      : '#fff'};
  font-size: ${props => getFontSize(props.size!)};
  padding: ${props => getPadding(props.size!)};
  border-radius: ${({ theme }) => theme.elements.roundness};
  border: none;

  box-sizing: border-box;

  font-weight: 500;
  font-stretch: normal;
  line-height: 1.4;
  margin: 0.2rem;
  display: relative;
  overflow: hidden;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  justify-content: center;
  align-items: center;

  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;

  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;

  transition: background-color 0.2s ease-in-out;
  will-change: background-color;

  :hover {
    background-color: ${props =>
      lightenOrDarken(get(props.theme.colors.goal, props.goal), -20)};
  }

  :active {
    background-color: ${props =>
      lightenOrDarken(get(props.theme.colors.goal, props.goal), -40)};
  }

  :focus {
    outline: none;
  }
`

Button.defaultProps = {
  goal: 'create',
  size: 'md',
  kind: 'default',
}

export default Button
