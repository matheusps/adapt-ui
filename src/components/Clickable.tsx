import * as React from 'react'
/** @jsx jsx */
import { jsx, css } from '@emotion/core'

import { useTheme } from '../hooks'
import { getColor } from '../helpers'

const Clickable: React.FC<ClickableType> = ({
  skin,
  children,
  full,
  disabled,
  onClick,
  ghost,
  noEffect,
  ...rest
}) => {
  const { elements, colors } = useTheme()

  const padding = '0.5rem'
  const margin = '0.5rem'

  const borderRadius = elements.roundness

  const selectedSkin = colors.skin[skin!]
  const color = getColor(selectedSkin, disabled ? 0.5 : 1)
  const bg = ghost
    ? 'transparent'
    : getColor(selectedSkin, disabled ? 0.08 : 0.1)
  const bgHover = !noEffect && getColor(selectedSkin, disabled ? 0.08 : 0.2)
  const bgActive = !noEffect && getColor(selectedSkin, 0.3)

  const handleClickEvent = (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    e.preventDefault()
    onClick && onClick(e)
  }

  return (
    <button
      {...rest}
      disabled={disabled}
      onClick={handleClickEvent}
      css={css`
        background-color: ${bg};
        color: ${color};
        padding: ${padding};
        border-radius: ${borderRadius};
        display: ${full ? 'block' : 'relative'};
        width: ${full ? '100%' : 'auto'};
        cursor: ${disabled ? 'not-allowed' : 'pointer'};
        margin: ${margin};

        border: none;
        box-sizing: border-box;
        font-weight: 500;
        font-stretch: normal;
        line-height: 1.4;
        overflow: hidden;
        text-align: center;
        text-decoration: none;
        text-transform: none;
        white-space: nowrap;
        justify-content: center;
        align-items: center;
        appearance: none;
        user-select: none;

        transition: background-color 0.2s ease-in-out;
        will-change: background-color;

        :hover {
          background-color: ${bgHover};
        }

        :active {
          background-color: ${bgActive};
        }

        :focus {
          outline: none;
        }
      `}
    >
      {children}
    </button>
  )
}

Clickable.defaultProps = {
  skin: 'primary',
  full: false,
}

export { Clickable }
