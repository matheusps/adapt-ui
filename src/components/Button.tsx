import React, { FC } from 'react'
import styled from 'styled-components'

import { getMeasure } from '../helpers'
import { Loader } from './Loader'
import { Clickable } from './core/Clickable'

const getFontSize = (measure: Measure): number =>
  getMeasure(measure, [0.8, 1, 1.25, 1.5])

const getPadding = (measure: Measure): string =>
  getMeasure(measure, [
    '0.375rem 1.4rem 0.375rem 1.4rem',
    '0.45rem 1.8rem 0.45rem 1.8rem',
    '0.65rem 2.2rem 0.65rem 2.2rem',
    '0.80rem 2.6rem 0.80rem 2.6rem',
  ])

const StyledClickable = styled(Clickable)<any>(({ fontSize, padding }) => ({
  fontSize,
  padding,
}))

const Button: FC<ButtonType> = ({ size, children, loading, ...props }) => {
  const fontSize = `${getFontSize(size!)}rem`
  const padding = getPadding(size!)

  return (
    <StyledClickable fontSize={fontSize} padding={padding} {...props}>
      {loading ? <Loader size={getFontSize(size!)} /> : children}
    </StyledClickable>
  )
}

Button.defaultProps = {
  size: 'md',
  loading: false,
}

export { Button }
