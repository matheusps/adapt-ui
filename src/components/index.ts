import { Button } from './Button'
import { IconButton } from './IconButton'
import { Emoji } from './Emoji'
import { Clickable } from './core/Clickable'
import { Flexible } from './Flexible'
import { Loader } from './Loader'
import { Surface } from './Surface'
import { Tabs } from './Tabs'
import { Tab } from './Tab'
import { Heading } from './Heading'
import { Paragraph } from './Paragraph'
import { DarksideTheme } from './DarksideTheme'
import { Icon } from './Icon'
import { FormGroup } from './FormGroup'
import { Form } from './Form'
import { Input } from './Input'
import { Label } from './Label'

export {
  Icon,
  Button,
  IconButton,
  Emoji,
  Clickable,
  Flexible,
  Loader,
  Surface,
  Tab,
  Tabs,
  Heading,
  Paragraph,
  DarksideTheme,
  FormGroup,
  Form,
  Input,
  Label,
}
