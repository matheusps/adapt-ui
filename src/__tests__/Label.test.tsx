import React from 'react'
import { render } from '@testing-library/react'

import { Label, DarksideTheme } from '../components'
import 'jest-styled-components'

describe('Label component', () => {
  const renderComponent = (customProps?) =>
    render(
      <DarksideTheme>
        <Label {...customProps!} />
      </DarksideTheme>
    )

  it('should render', () => {
    const component = renderComponent()
    expect(component).toBeTruthy()
  })

  it('should match snapshot as default', () => {
    const component = renderComponent()
    expect(component.asFragment()).toMatchSnapshot()
  })
})
