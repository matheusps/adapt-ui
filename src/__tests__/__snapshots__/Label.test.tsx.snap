// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should match snapshot as default 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1.1rem;
  display: block;
  line-height: 1.2rem;
}

<label
    class="c0"
  />
</DocumentFragment>
`;
